ServerCore/Pages/Events/Players.cshtml.cs

```diff
@@ -12,28 +13,108 @@ namespace ServerCore.Pages.Events
 {
     public class PlayersModel : EventSpecificPageModel
     {
+        public IList<PuzzleUser> Admins { get; set; }
+
+        public string AdminEmails { get; set; }
+
+        public IList<Tuple<PuzzleUser, int>> Authors { get; set; }
+
+        public string AuthorEmails { get; set; }
+
         public IList<TeamMembers> Players { get; set; }
 
-        public string Emails { get; set; }
+        public string PlayerEmails { get; set; }
 
         public PlayersModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
         {
         }
 
-        public async Task<IActionResult> OnGetAsync(int teamId)
+        public async Task<IActionResult> OnGetAsync()
         {
+            // Get admins
+
+            Admins = await _context.EventAdmins
+                .Where(admin => admin.Event == Event).Select(admin => admin.Admin)
+                .ToListAsync();
+
+            StringBuilder adminEmailList = new StringBuilder("");
+            foreach (PuzzleUser admin in Admins)
+            {
+                adminEmailList.Append(admin.Email + "; ");
+            }
+            AdminEmails = adminEmailList.ToString();
+
+            // Get authors
+
+            IList<PuzzleUser> allAuthors = await (from author in _context.EventAuthors
+                                                  where author.Event == Event
+                                                  select author.Author).ToListAsync();
+
+            // note - if there's a faster way to count the puzzles / join all this data please let me know - Jenna
+            IList<IGrouping<int, PuzzleAuthors>> allPuzzles = await (from puzzleAuthor in _context.PuzzleAuthors
+                                                     where puzzleAuthor.Puzzle.Event == Event
+                                                     group puzzleAuthor by puzzleAuthor.Author.ID into puzzleGroup
+                                                     select puzzleGroup)
```

Comment:
You can do something like:
select new {Name = puzzleGroup.Key, Count = puzzlegroup.Count()}

You can then enumerate all authors and compare that to these results to find any that came back as zero. For an example, take a look at Teams/List.cshtml.cs

Reply:
Done :D Thank you :)